spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/MethodParameters.groovy

```diff
@@ -176,4 +176,17 @@ def foo(x, y) {
     SpockExecutionException see = thrown()
     see.message == /No argument was provided for parameters: 'x', 'y'/
   }
+
+  @Issue("https://github.com/spockframework/spock/issues/1279")
+  def "data values are casted with type-coercion"(Integer x, Integer y) {
```

Comment:
Looking at code, it should also work for `int`. Maybe you could change `y` to `int y` to cover it by a test?

Reply:
Sure I can change that, but the type will still be `Integer`
If you type in a Groovy console
```groovy
int i = 1
i.getClass()
```
you also get `Integer` as result.
It works with any type coercion, even custom ones. :-)

Reply:
Sure I can change that, but the type will still be `Integer`
If you type in a Groovy console
```groovy
int i = 1
i.getClass()
```
you also get `Integer` as result.
It works with any type coercion, even custom ones. :-)

Reply:
Yes, I know. I meant, put `int y` in the method declaration - which is usually more natural for people than using `Integer` - to have regression tests verifying also that.